### PR TITLE
Updates to support groovy 4

### PR DESCRIPTION
**Pull request by AngMits**
## Fixes

- Parsing Groovy 4 switch statements.
- I have been facing issues with parsing switch statements after upgrading work-related projects to Groovy 4. The current state of the linter will throw a parser error if it encounters such a statement.
- Instead of adding exclusions for each case, adding support for the language would be a slightly better alternative.

## Proposed Changes

- Update jars to their latest versions (commons)
- Update jars to support Groovy 4

### Additional info ℹ️ 

- Although local tests are passing, and a trial run using the actions showed that nothing is broken, please request any changes that might be required or any other course of action.
- The added import in the `CodeNarcServer` was required, otherwise Groovy would throw an error.
- The update of the workflow file was done to check the workflows, otherwise, an error version would be thrown.

### CODE REVIEW

**Comment by nvuillam** [comment]
Amazing, i'll test tht asap :)

cc @stevenh

**Comment by stevenh** [comment]
Just a quick note to say I've seen this and its on my list to look at.

File: groovy/src/main/com/nvuillam/CodeNarcServer.groovy [hunk 1]
```diff
@@ -240,6 +243,6 @@ class CodeNarcServer {
         server.stop(1)
         LOGGER.info('Stopped')
         latch.countDown()
-   }
+    }
```
Comment (stevenh) [inline groovy/src/main/com/nvuillam/CodeNarcServer.groovy:246]: question: This looks like an editor format change that isn't needed?

Comment (AngMits) [reply groovy/src/main/com/nvuillam/CodeNarcServer.groovy:246]: Looks like it, will revert to avoid unwanted changes here. ⚙️

Comment (AngMits) [reply groovy/src/main/com/nvuillam/CodeNarcServer.groovy]: Fixed, thanks!

**Review by stevenh — APPROVED** [review]
LGTM

**Pull request merged**